lib/fog/xen_server/connection.rb

```diff
@@ -5,8 +5,11 @@ module XenServer
     class Connection
       attr_reader :credentials
 
-      def initialize(host, timeout)
-        @factory = XMLRPC::Client.new(host, "/")
+      def initialize(host, port, use_ssl, timeout)
```

Comment:
This signature is bothering me. `use_ssl` can be either `true`, `false` or `-1`.

What if `use_ssl` accepts only `true` or `false` and we support another parameter called `verify_mode` with a sane default?
The signature would look like `initialize(host, port, use_ssl, verify_mode, timeout)`. WDYT?